DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetAccountManagerWindow.cs

```diff
@@ -0,0 +1,139 @@
+            var btnLogout = new XNAClientButton(WindowManager);
+            btnLogout.Name = "btnLogout";
+            btnLogout.ClientRectangle = new Rectangle(Width - 104,
+                btnConnect.Y, 92, 23);
+            btnLogout.Text = "Logout";
+            btnLogout.LeftClick += BtnLogout_LeftClick;
+
+            ddAccounts = new XNADropDown(WindowManager);
+            ddAccounts.Name = "ddAccounts";
+            ddAccounts.Text = "Accounts";
+            ddAccounts.ClientRectangle = new Rectangle(100, ClientRectangle.Y + 50, 200, 19);
+
+            lblAccounts = new XNALabel(WindowManager);
+            lblAccounts.Name = "lblAccounts";
+            lblAccounts.FontIndex = 1;
+            lblAccounts.Text = "Nicknames:";
+            lblAccounts.ClientRectangle = new Rectangle(12, ddAccounts.ClientRectangle.Y + 1,
+            lblAccounts.ClientRectangle.Width, lblAccounts.ClientRectangle.Height);
+
+            AddChild(ddAccounts);
+            AddChild(lblAccounts);
+            AddChild(btnConnect);
+            AddChild(btnLogout);
+
+            base.Initialize();
+
+            CenterOnParent();
+
+            Keyboard.OnKeyPressed += Keyboard_OnKeyPressed;
+
+            CnCNetAPI.Instance.AccountUpdated += CnCNetAuthApi_AccountUpdated;
+        }
+
+        private void CnCNetAuthApi_AccountUpdated(object obj)
+        {
+            PopulateAccountList();
+        }
+
+        private void Keyboard_OnKeyPressed(object sender, KeyPressEventArgs e)
+        {
+            if (Enabled && e.PressedKey == Keys.Enter)
+                BtnConnect_LeftClick(this, EventArgs.Empty);
+        }
+
+        private void BtnLogout_LeftClick(object sender, EventArgs e)
+        {
+            CnCNetAPI.Instance.Logout();
+            Disable();
+        }
+
+        private void BtnConnect_LeftClick(object sender, EventArgs e)
+        {
+            if (string.IsNullOrEmpty(ddAccounts.SelectedItem.Text))
+                return;
+
+            string nickname = ddAccounts.SelectedItem.Text;
+
+            string errorMessage = NameValidator.IsNameValid(nickname);
+
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                XNAMessageBox.Show(WindowManager, "Invalid Player Name", errorMessage);
+                return;
+            }
+
+            ProgramConstants.PLAYERNAME = nickname;
+            UserINISettings.Instance.PlayerName.Value = ProgramConstants.PLAYERNAME;
+            UserINISettings.Instance.SaveSettings();
+
+            Connect?.Invoke(this);
+        }
+
+        private void PopulateAccountList()
+        {
+            for (int i = 0; i < CnCNetAPI.Instance.Accounts.Count; i++)
+            {
+                AuthPlayer player = CnCNetAPI.Instance.Accounts[i];
+
+                XNADropDownItem ddItem = new XNADropDownItem();
+                ddItem.Text = player.username;
+
+                ddAccounts.AddItem(ddItem);
+            }
+            ddAccounts.SelectedIndex = 0;
+        }
```

Comment:
Should you clear the account list in the drop down before the loop? If the user logs in once, then logs out and then starts logging in again, do they get their accounts duplicated?

If CNCNetAPI.Instance.Accounts.Count can be 0, you should check for it before setting SelectedIndex = 0.